apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/SurveyEditor.tsx

```diff
@@ -37,6 +38,20 @@ export default function SurveyEditor({ environmentId, surveyId }: SurveyEditorPr
     }
   }, [survey]);
 
+  useEffect(() => {
```

Comment:
Would it be possible to move this code to `SurveyMenuBar.tsx` and use the same modal you used for the back button, but with different callbacks on close/save? 
I think having all the logic in the same place would be great, and if it's also possible to have the nice modal as the default browser confirmation, even better :-)

Reply:
sure, will move the logic to `SurveyMenuBar.tsx`. 👍

But removing the default browser confirmation dialog for `beforeunload` is not possible I think.
In my research, I also found that we can show a custom dialog but the default dialog will still be present.
Do you have any ideas or suggestions's how we can achieve this without the default browser dialog?

Reply:
@joe-shajan Yeah, you are right. Then we stick with the current built-in browser modal 😊